ecosystem/typescript/sdk/src/plugins/coin_client.ts

```diff
@@ -54,8 +61,32 @@ export class CoinClient {
       // If this is the first time an account has received the specified coinType,
       // and this is set to false, the transaction would fail.
       createReceiverIfMissing?: boolean;
+      // fungible asset address if you wish to transfer a fungible asset
+      assetAddress?: MaybeHexString;
     },
   ): Promise<string> {
+    if (extraArgs?.assetAddress) {
+      /* eslint-disable no-console */
+      console.warn("to transfer a fungible asset, use `FungibleAssetClient()` class for a better support");
+      const provider = new Provider({
+        fullnodeUrl: this.aptosClient.nodeUrl,
+        indexerUrl: NetworkToIndexerAPI[NodeAPIToNetwork[this.aptosClient.nodeUrl]] ?? this.aptosClient.nodeUrl,
+      });
+      const fungibleAsset = new FungibleAssetClient(provider);
+      let recipient = to;
```

Comment:
This can be removed by moving this line up from below:

const toAddress = getAddressFromAccountOrAddress(to);